cmd/embedded-cluster/list_images.go

```diff
@@ -2,27 +2,47 @@ package main
 
 import (
 	"fmt"
+	"os"
 
+	ecv1beta1 "github.com/replicatedhq/embedded-cluster/kinds/apis/v1beta1"
 	"github.com/replicatedhq/embedded-cluster/pkg/config"
+	"github.com/replicatedhq/embedded-cluster/pkg/defaults"
 	"github.com/urfave/cli/v2"
 )
 
-var listImagesCommand = &cli.Command{
-	Name:   "list-images",
-	Usage:  "List images embedded in the cluster",
-	Hidden: true,
-	Action: func(c *cli.Context) error {
-		k0sCfg := config.RenderK0sConfig()
+func listImagesCommand() *cli.Command {
+	runtimeConfig := ecv1beta1.GetDefaultRuntimeConfig()
 
-		metadata, err := gatherVersionMetadata(k0sCfg)
-		if err != nil {
-			return fmt.Errorf("failed to gather version metadata: %w", err)
-		}
+	return &cli.Command{
+		Name:   "list-images",
+		Usage:  "List images embedded in the cluster",
+		Hidden: true,
+		Flags: []cli.Flag{
+			getDataDirFlag(runtimeConfig),
+		},
+		Before: func(c *cli.Context) error {
+			if os.Getuid() != 0 {
+				return fmt.Errorf("version metadata command must be run as root")
+			}
+			return nil
```

Comment:
I still don't really understand why this takes sudo now

Reply:
its cause we have to create the tmp dir in case anything uses it when we instantiate the provider. could i fast follow with trying to remove the sudo dependency?